contracts/src/CrossChainApplications/examples/NativeTokenBridge/NativeTokenDestination.sol

```diff
@@ -213,27 +280,47 @@ contract NativeTokenDestination is TeleporterOwnerUpgradeable, INativeTokenDesti
 
         (address recipient, uint256 amount) = abi.decode(message, (address, uint256));
         require(recipient != address(0), "NativeTokenDestination: zero recipient address");
-        require(amount != 0, "NativeTokenDestination: zero transfer value");
+
+        uint256 scaledAmount = _scaleTokens(amount, true);
+        require(scaledAmount >= 0, "NativeTokenDestination: zero scaled amount received");
 
         // If the contract has not yet been collateralized, we will deduct as many tokens
         // as needed from the transfer as needed. If there are any excess tokens, they will
         // be minted and sent to the recipient.
-        uint256 adjustedAmount = amount;
+        uint256 adjustedAmount = scaledAmount;
         if (currentReserveImbalance > 0) {
-            if (amount > currentReserveImbalance) {
+            if (scaledAmount > currentReserveImbalance) {
                 emit CollateralAdded({amount: currentReserveImbalance, remaining: 0});
-                adjustedAmount = amount - currentReserveImbalance;
+                adjustedAmount = scaledAmount - currentReserveImbalance;
                 currentReserveImbalance = 0;
             } else {
-                currentReserveImbalance -= amount;
-                emit CollateralAdded({amount: amount, remaining: currentReserveImbalance});
-                return;
+                emit CollateralAdded({amount: scaledAmount, remaining: currentReserveImbalance});
```

Comment:
Are we emitting the stale `currentReserveImbalance` here now prior to updating it? Seems like we'd want to emit the updated value.

Reply:
Yeah this was broken, fixed now.